sdmetrics/single_table/detection/base.py

```diff
@@ -77,19 +77,15 @@ def compute(cls, real_data, synthetic_data, metadata=None):
         if np.isin(X, [np.inf, -np.inf]).any():
             X[np.isin(X, [np.inf, -np.inf])] = np.nan
 
-        try:
-            scores = []
-            kf = StratifiedKFold(n_splits=3, shuffle=True)
-            for train_index, test_index in kf.split(X, y):
-                y_pred = cls._fit_predict(X[train_index], y[train_index], X[test_index])
-                roc_auc = roc_auc_score(y[test_index], y_pred)
-
-                scores.append(max(0.5, roc_auc) * 2 - 1)
-
-            return 1 - np.mean(scores)
-        except ValueError as err:
```

Comment:
It seems like `compute` should throw errors, and then we can catch them in `compute_metrics` and process the error as desired (e.g. continue onto other metrics)